lib/Parse/ParseExpr.cpp

```diff
@@ -2753,17 +2753,49 @@ void Parser::ParseBlockId(SourceLocation CaretLoc) {
 }
 
 bool Parser::StartsBoundsExpression(Token &tok) {
-  if (Tok.getKind() == tok::identifier) {
+  if (tok.getKind() == tok::identifier) {
     IdentifierInfo *Ident = Tok.getIdentifierInfo();
     return (Ident == Ident_byte_count || Ident == Ident_count ||
             Ident == Ident_bounds);
   }
   return false;
 }
 
+bool Parser::StartsInteropTypeAnnotation(Token &tok) {
+  switch (tok.getKind()) {
+    case tok::kw__Ptr:
+    case tok::kw__Array_ptr:
```

Comment:
We'll want to keep this list in sync with the full list of all checked-c pointer types (including spans and whatever else we add).  Is there already a helper function elsewhere that checks if a given token is a checked-c type token?